village/src/components/SmurfForm.js

```diff
@@ -34,20 +49,24 @@ class SmurfForm extends Component {
             placeholder="name"
             value={this.state.name}
             name="name"
+            required
           />
           <input
             onChange={this.handleInputChange}
             placeholder="age"
             value={this.state.age}
             name="age"
+            type='number'
+            required
           />
           <input
             onChange={this.handleInputChange}
             placeholder="height"
             value={this.state.height}
             name="height"
+            required
           />
-          <button type="submit">Add to the village</button>
+          <button onClick={this.addSmurf} type="submit">Add to the village</button>
```

Comment:
you don't need an onClick handler, this button will cause form submit, because it is type="submit", and form has onSubmit set to this.addSmurf already